tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -0,0 +1,379 @@
+                }
+            }
+
+            // TODO: Add container tags from attributes if the tag isn't already in the span
+
+            // Fixup "env" tag
+            if (span.Context.TraceContext?.Environment is null
+                && span.GetTag("deployment.environment") is string otelServiceEnv
+                && !string.IsNullOrEmpty(otelServiceEnv))
+            {
+                span.SetTag(Tags.Env, otelServiceEnv);
+            }
+
+            // TODO: The .NET OTLP exporter doesn't currently add tracestate, but the DD agent will set it as tag "w3c.tracestate" if detected
+            // For now, we can keep this unimplemented so the resulting span matches the .NET OTLP exporter
+            // span.SetTag("w3c.tracestate", w3CActivity.TraceStateString);
+
+            // Add the library name and library version
+            if (activity5 is not null)
+            {
+                span.SetTag("otel.library.name", activity5.Source.Name);
+                span.SetTag("otel.library.version", activity5.Source.Version);
+            }
+
+            // Set OTEL status code and OTEL status description
+            if (span.GetTag("otel.status_code") is null)
+            {
+                span.SetTag("otel.status_code", "STATUS_CODE_UNSET");
+            }
+
+            // Map the OTEL status to error tags
+            AgentStatus2Error(activity, span);
+
+            if (span.OperationName is null)
+            {
+                if (activity5 is not null)
+                {
+                    // Later: Support config 'span_name_as_resource_name'
+                    // Later: Support config 'span_name_remappings'
+                    span.OperationName = activity5.Source.Name switch
+                    {
+                        string libName when !string.IsNullOrEmpty(libName) => $"{libName}.{SpanKindNames[(int)activity5.Kind]}",
+                        _ => $"opentelemetry.{SpanKindNames[(int)activity5.Kind]}",
+                    };
+                }
+                else
+                {
+                    span.OperationName = activity.OperationName;
+                }
+            }
+
+            // Update Service with a reasonable default
+            if (span.ServiceName is null)
+            {
+                span.ServiceName = span.GetTag("peer.service") switch
+                {
+                    string peerService when !string.IsNullOrEmpty(peerService) => peerService,
+                    _ => "OTLPResourceNoServiceName",
+                };
+            }
+
+            // Update Resource with a reasonable default
+            if (span.ResourceName is null)
+            {
+                // TODO: Implement resourceFromTags
+                // See: https://github.com/DataDog/datadog-agent/blob/67c353cff1a6a275d7ce40059aad30fc6a3a0bc1/pkg/trace/api/otlp.go#L555
+                span.ResourceName = activity5?.DisplayName;
```

Comment:
After looking through the code, the `DisplayName` should never be null. If the user updates `DisplayName` using the setter then that value will be returned. Otherwise, the value of `OperationName` will be returned, which will never be null

Reply:
Perfect, thanks a lot for checking